### ISSUE TITLE
Unable to update volume labels

### ISSUE THREAD

**Issue by mcanevet** [docker/docker#25291]
Hello,

It looks like there is no way to update volume labels or am I missing something?
Volume API looks to support only supports listing, creating (with labels), inspect and remove a volume.

Cheers,
Mickaël

**Comment by thaJeztah** [comment]
Correct, updating existing volumes is currently not supported; it may be something we could consider, although we need to carefully look what properties can be updated for an existing volume, and which options are "immutable"

Changing / updating labels does sound like a candidate to allow to be updated (if we decide on implementing)

/cc @cpuguy83 @ehazlett

**Comment by allencloud** [comment]
I think it is a useful feature for volume management. But first, like @thaJeztah mentioned, we need to do a detailed plan for this.

**Comment by govint** [comment]
Plus, being able to grow (especially) or shrink a disk, storage policy for example. And for persistent volumes backed by volume drivers it could include virtually any property that the volume driver can manipulate.

**Comment by ewestfal** [comment]
Curious if there is any further thought to adding this capability (for networks as well). We use Docker EE and it uses labels for access control (to identify collections). Not being able to update those is really painful and impactful when we need to refactor our permissions or transfer volumes/networks between collections.

**Comment by cpuguy83** [comment]
I think it's fine to support updating labels. Does someone want to work on it?

**Comment by jmartin84** [comment]
Is this still something that needs to be done?

**Comment by thaJeztah** [comment]
Yes, I don't think someone worked in this; I think this only requires changes in the CLI (as the API side changes are already in); https://github.com/docker/cli
